sktime/forecasting/var_reduce.py

```diff
@@ -0,0 +1,445 @@
+            If returned as a DataFrame, the columns are named according to the original
+            column names with the lag number appended (e.g., 'A_lag1', 'B_lag2').
+        """
+        # Take the last lags rows of the input data
+        if isinstance(data, np.ndarray):
+            data = pd.DataFrame(data, columns=self.var_names)
+        lags = self.lags
+        lagged_data = data.tail(lags)
+
+        # Invert the data
+        inverted_data = lagged_data[::-1]
+
+        # Linearize it
+        linearized_data = inverted_data.values.flatten().reshape(1, -1)
+
+        # Return as np.array or pd.DataFrame
+        if return_as_ndarray:
+            return np.array(linearized_data)
+        else:
+            # Generate column names
+            columns = []
+            for lag in range(1, lags + 1):
+                columns += [f"{col}_lag{lag}" for col in data.columns]
+            return pd.DataFrame(linearized_data, columns=columns)
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data.
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            Guaranteed to have a single column if scitype:y=="univariate".
+        fh : ForecastingHorizon, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous time series to fit to; will be ignored
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from copy import deepcopy
+
+        var_names = y.columns
+        self.var_names = var_names
+
+        X, Y = self._prepare_for_fit(y, return_as_ndarray=False)
+        n, k = X.shape
+        num_series = y.shape[1]
+        self.num_series = num_series
+
+        # Initialize placeholders for coefficients and intercepts
+        coefficients = np.zeros((self.lags, num_series, num_series))
+        intercepts = np.zeros(num_series)
+
+        # Try multioutput fitting
+        try:
+            model = deepcopy(self.regressor_)
+            model.fit(X, Y)
+            self.multioutput_regressor = model
+            # if the model has `.intercept_` and `.coef_` attributes, extract them
+            if hasattr(model, "intercept_") and hasattr(model, "coef_"):
+                self.coefficients_ = model.coef_.reshape(
+                    (self.num_series, self.lags, self.num_series)
+                )
+                self.coefficients_ = np.transpose(self.coefficients_, (1, 0, 2))
+                self.intercept_ = model.intercept_
+        except ValueError:
+            # Exception means regressor doesn't support multioutput;
+            # for each target series `y` in `Y`, a separate regressor is
+            # trained using `y` and the common predictors `X`.
+            # TODO: use sklearn's MultiOutputRegressor to unify API
+            for i, var_name in enumerate(var_names):
```

Comment:
what I would do instead, if `y` is multivariate, is to wrap the regressor in `MultiOutputRegressor`, and write it to `regressor_`.

That way, `regressor_` has no case-by-case differring type, and is always the fitted regressor.

It should also solve the remaining failures related to `get_fitted_params`, which encounters an unfitted `regressor_`.

Reply:
(Addressed and resolved)